File: exercism/go/rna-transcription/rna_transcription.go
// https://exercism.io/my/solutions/e19201440e64447b8467d42ae88bf21f

package strand

import "strings"

var complement = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

// ToRNA transforms a DNA string into an RNA one
func ToRNA(dna string) string {
	return strings.Map(func(nucleotide rune) rune { return complement[nucleotide] }, dna)
}

File: exercism/elixir/rna-transcription/lib/rna_transcription.ex
# https://exercism.io/my/solutions/6a447d0395294e1fab9d686cfc8f6801

defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    trans = %{?A => ?U, ?C => ?G, ?G => ?C, ?T => ?A}
    Enum.map(dna, fn c -> trans[c] end)
  end
end
